controller/iteration_blackbox_test.go

```diff
@@ -1239,6 +1239,43 @@ func (rest *TestIterationREST) TestUpdateIteration() {
 		svc, ctrl := rest.SecuredControllerWithIdentity(fxt2.Identities[0])
 		test.UpdateIterationForbidden(t, svc.Context, svc, ctrl, beta.ID.String(), &payload)
 	})
+
+	rest.T().Run("update fail - new parent is one of child", func(t *testing.T) {
+		// build following structure
+		// 	root 0
+		// 		|---- itr 1
+		// 			|---- itr 2
+		// 				|---- itr 3
+		//  					|---- itr 4
+
+		// given
+		fxt := tf.NewTestFixture(t, rest.DB,
+			tf.Iterations(5, tf.SetIterationNames("root iteration", "iteration 1",
+				"iteration 2", "iteration 3", "iteration 4"),
+				func(fxt *tf.TestFixture, idx int) error {
+					itr := fxt.Iterations[idx]
```

Comment:
This can be simplified to:

```go
func(fxt *tf.TestFixture, idx int) error {
    if idx > 0 {
        fxt.Iterations[idx].MakeChildOf(*fxt.Iterations[idx-1])
    }
    return nil
}
```

Reply:
yes done in https://github.com/fabric8-services/fabric8-wit/pull/1770/commits/4a007fb48f8f08955773a5da0a4638134e26b6b8